### PR TITLE
MEPTS-390: Changes on A1 and A2

### PR DESCRIPTION
**Pull request by simaomazive**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/metadata/HivMetadata.java [hunk 1]
```diff
@@ -1480,4 +1480,16 @@ public Concept getPreTarvConcept() {
         Context.getAdministrationService().getGlobalProperty("eptsreports.preTarvConceptUuid");
     return getConcept(uuid);
   }
+
+  /**
+   * Concept_id = 23891
+   *
+   * @return
+   */
+  public Concept getDateOfMasterCardFileOpening() {
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/metadata/HivMetadata.java]: @simaomazive please put concept at the end of the name so we can identify quickly

Comment (psmatsinhe) [reply api/src/main/java/org/openmrs/module/eptsreports/metadata/HivMetadata.java]: Like 

> **getDateOfMasterCardFileOpeningConcept()**

**Review by psmatsinhe — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/cohort/evaluator/EptsTransferredInCohortDefinitionEvaluator.java [hunk 2]
```diff
@@ -63,17 +64,21 @@ public EvaluatedCohort evaluate(CohortDefinition cohortDefinition, EvaluationCon
     q.append("       AND transf.voided = 0 ");
     q.append("       AND transf.concept_id = :transferFromOther ");
     q.append("       AND transf.value_coded = :yes ");
+
+    q.append("       AND opening.voided = 0 ");
+    q.append("       AND opening.concept_id = :dateOfMasterCardFileOpening ");
+
     if (cd.getOnOrAfter() == null) {
-      q.append("     AND transf.obs_datetime < :onOrBefore ");
+      q.append("     AND opening.value_datetime < :onOrBefore ");
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/cohort/evaluator/EptsTransferredInCohortDefinitionEvaluator.java:74]: @simaomazive switch position of this

**Review by psmatsinhe — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/eptsreports/metadata/HivMetadata.java (hunk 1 above)
Comment (simaomazive) [reply api/src/main/java/org/openmrs/module/eptsreports/metadata/HivMetadata.java]: Done.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/cohort/evaluator/EptsTransferredInCohortDefinitionEvaluator.java (hunk 2 above)
Comment (simaomazive) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/cohort/evaluator/EptsTransferredInCohortDefinitionEvaluator.java:74]: Done

File: omod/src/main/resources/config.xml [hunk 3]
```diff
@@ -351,10 +351,10 @@
 	</globalProperty>
 
 	<globalProperty>
-		<property>@MODULE_ID@.dateOfMasterCardFileOpeningUuid</property>
+		<property>@MODULE_ID@.dateOfMasterCardFileOpeningConceptUuid</property>
```
Comment (psmatsinhe) [inline omod/src/main/resources/config.xml:354]: @simaomazive please leave this as uuid

> dateOfMasterCardFileOpeningUuid

**Review by psmatsinhe — CHANGES_REQUESTED** [review]

File: omod/src/main/resources/config.xml (hunk 3 above)
Comment (psmatsinhe) [reply omod/src/main/resources/config.xml:354]: > dateOfMasterCardFileOpeningConceptUuid

**Review by psmatsinhe — APPROVED** [review]

**Pull request merged**